### DISCUSSION TITLE
Different way to specify child nodes?

### DISCUSSION THREAD

**Discussion by ericsampson** [kdl-org/kdl#66] [Ideas]
I love everything about the way KDL files look, except for the use of curly braces to define child nodes. The presence of braces is really jarring in the document, to me it really sticks out as different than the appearance or "feel" of the rest of the language.

Does anyone else feel similar?

If so, does anyone have any thoughts on alternate approaches to delineate child nodes?

Cheers,
Eric

**Comment by zkat** [comment]
What would you suggest?

**Reply by ericsampson** [reply]
What about pipes, parens, or underscores?

I think what makes the curlies jarring to me is that it triggers my "this is JSON" neurons, because it's so burnt into my brain - and so it causes a confusing context switch.

**Reply by ericsampson** [reply]
pipes:

```
contents |
  section "First section" |
    paragraph "This is the first paragraph"
    paragraph "This is the second paragraph"
  |
|
```

**Reply by ericsampson** [reply]
parens:

```
contents (
  section "First section" (
    paragraph "This is the first paragraph"
    paragraph "This is the second paragraph"
  )
)
```

**Reply by ericsampson** [reply]
underscore:

```
contents _
  section "First section" _
    paragraph "This is the first paragraph"
    paragraph "This is the second paragraph"
  _
_
```

**Reply by ericsampson** [reply]
Personally, after seeing these examples, I like parens quite a bit. The round nature communicates nesting, and it's similar enough to JSON to feel familiar, but also just different enough to avoid triggering the response I mentioned before. 

What do you think?

**Reply by ericsampson** [reply]
The roundness of the parens also feels "kuddly", compared to the sharp shape of brackets or the austere shape of pipes :)

**Reply by hkolbeck-streem** [reply]
Of all these options, I think parens is the most feasible. Having opposing start and end characters makes things easier to read in my opinion. That said, I don't view curly brackets negatively, and would prefer to stick with them.
